### ISSUE TITLE
Ontologies & SHACL-shapes for OpenLABEL

### ISSUE THREAD

**Issue by rcrswld** [GAIA-X4PLC-AAD/ontology-management-base#38]
We need to create both ontology and SHACL in order to use OpenLABEL for our extended metadata.

Right now, I already created an ontology-draft for OpenLABEL in separate branch, s.:
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/openlabel-ref/openlabel/openlabel_ref_ontology_extended.ttl

The SHACL-shape is yet missing.

@heuerfin, I saw your comments and suggestion in scenario pull request https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/6

As this pull request is dedicated to our custom extensions from PLC-AAD or SRMD/SetLevel I would like to keep OpenLABEL out there for a moment. Can we further elaborate OpenLABEL specific issues here and the openlabel_ref branch from above?

Generally speaking, my suggestion would be to nest OpenLABEL under scenario-metadata but separate to SRMD-extension and our PLC-AAD specific content. Therefore, it seems easier to me reusing OpenLABEL-parts also for other domains like maps and sensor-data.

What would you think?

**Comment by heuerfin** [comment]
Yes, I agree that we should keep them seperate. Currently, the main branch contains a mix of 4 PLC specific and SET Level attributes. We should clean that one up.

One question: can leave out the duplicate file like you added in https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/openlabel-ref/openlabel/openlabel_ref_ontology_extended.ttl and just import it like
```
@prefix openlabel: <https://openlabel.asam.net/V1-0-0/ontologies/> .
@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .

scenario: a owl:Ontology ;
    rdfs:label "Ontology definition for scenario"@en ;
    dcterms:contributor "Fin Heuer (DLR), Rico Auerswald (FhG IVI)" ;
    dcterms:creator "The GAIA-X 4 PLC AAD Project Team" ;
    dcterms:identifier "https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/" ;
    owl:versionIRI <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/scenario_ontology.ttl> ;
    owl:versionInfo "0.1 ;
    owl:imports <https://openlabel.asam.net/V1-0-0/ontologies/openlabel_ontology_scenario_tags.ttl> .

scenario:Scenario a owl:Class ;
    rdfs:label "GXScenario"@en ;
    rdfs:comment "Class definition for a scenario"@en ;
    rdfs:subClassOf gx:DataResource ,
                    openlabel:Scenario .
```

The `owl:imports <https://openlabel.asam.net/V1-0-0/ontologies/openlabel_ontology_scenario_tags.ttl> .` part is currently not merged. In Protege it looks quiet fine to also include the OpenLABEL ontology.

**Comment by rcrswld** [comment]
In alignment with @robertschubert in AP 1.4 -> OpenLABEL-ontology will be stored in ontology-mngt.-base repo

**Comment by BugGreen** [comment]
## PROPOSAL: Integrating OpenLABEL into Scenario and HD Map Ontologies

### 1. Context & Goals

* **envited-x Ontology** defines a general `SimulationAsset` with properties like `hasDataResource`, `hasDataResourceExtension`, and `hasManifest`.
* **OpenLABEL Ontology** defines domain‑specific classes for scenario description (e.g. `openlabel:Scenario`, `openlabel:OddScenery`, `openlabel:Tag`).
* **Scenario Ontology** and **HD Map Ontology** currently subclass `envited-x:SimulationAsset` to describe scenario and HD map assets respectively.
* **Objective:** Seamlessly integrate OpenLABEL concepts into both ontologies so that their `hasContent` properties can carry either native content or OpenLABEL content.

---

### 2. Current Class Hierarchies

| Class              | SubclassOf                                    | Effect                                                                     |
| ------------------ | --------------------------------------------- | -------------------------------------------------------------------------- |
| scenario\:Scenario | envited-x\:SimulationAsset and openlabel\:Scenario | Inherits both generic simulation‑asset rules and OpenLABEL scenario rules. |
| hdmap\:HdMap       | envited-x\:SimulationAsset                    | Inherits only the generic simulation‑asset rules.                          |

### 3. Why and how - UnionOf on `hasContent`?

> **Proposal in a nutshell**  

> Instead of making every asset class (e.g. `hdmap:HdMap`, `scenario:Scenario`) a direct subclass of an OpenLABEL class, we attach OpenLABEL content only where it is needed—through a helper class that uses **`owl:unionOf`** and is referenced as the **range** of the property `…hasContent`.

##### 3.1 What `owl:unionOf` is  
`owl:unionOf` is an OWL 2 class-constructor defined in the W3C **OWL 2 Structural Specification & Direct Semantics** §2.1.1 (“Class Expressions”) which states that the union “contains exactly the individuals that belong to *at least one* of the operand classes.”<sup>1</sup>  
It lets us declare a new class whose instances can be of *either* class A *or* class B (or C …).

```turtle
# Example of helper class

hdmap:ContentOrOddScenery a owl:Class ;

  owl:unionOf ( hdmap:Content openlabel:OddScenery ) .
```

The above example means that any node of type hdmap:ContentOrOddScenery is therefore inferred to be 
hdmap:Content or openlabel:OddScenery.

#### 3.2 Why use the union-range helper pattern?

| Goal                                      | Achieved with union-range helper                                                                        | Problem with direct subclassing                                                                             |
| ----------------------------------------- | ------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
| **Keep asset semantics clean**            | Only the `hasContent` slot is polymorphic; `HdMap` remains purely a map, `Scenario` remains a scenario. | `HdMap ⊑ openlabel:OddScenery` would assert that *every* HD map is an OddScenery object—conceptually wrong. |
| **Minimal impact when OpenLABEL evolves** | Changes isolated to the helper class or SHACL shape.                                                    | Any new OpenLABEL restriction would automatically apply to every HdMap instance.                            |
| **Straightforward validation**            | SHACL `sh:or` can validate “must conform to either shape A or shape B”.                                 | Mixed hierarchies complicate shape design and debugging.                                                    |
| **Ease of reasoning**                     | Reasoners only infer the extra type on the linked content node.                                         | Reasoners infer extra types/properties on the *asset* itself (risk of unexpected entailments).              |

#### How it works (step-by-step)

##### 1. Create a helper class
```turtle
scenario:ContentOrTag a owl:Class ;
  owl:unionOf ( scenario:Content openlabel:Tag ) .
```
##### 2. Point the property range to its
```turtle
scenario:hasContent rdfs:range scenario:ContentOrTag .
```
##### 3. Add SHACL validation
```turtle
sh:property [
  sh:path scenario:hasContent ;
  sh:or ( [ sh:node scenario:ContentShape ]
          [ sh:node openlabel:TagShape ] ) ;
  sh:message "Content must be Scenario-Content or OpenLABEL Tag." ;
] .
```
##### 4. Leave the asset classes alone

No change to `hdmap:HdMap` or `scenario:Scenario` superclasses: they stay `envited-x:SimulationAsset`.

**Result:** Consumers of each property can supply either form of content and still satisfy OWL range-checking.

---

### 4. Summary

* **UnionOf ranges** let a single property carry multiple content types without altering class hierarchies.
* **Scenario** and **HdMap** remains only `envited‑x` to preserve domain semantics; union‑range handles OpenLABEL integration.
* **SHACL** shapes finalize the contract by validating the actual content structure.

This design shall integrate OpenLABEL into Scenario and HDMap ontologies while preserving clear, maintainable class definitions.


<sup>1</sup> Source: W3C OWL 2 Specification, §2.1.1 “Class Expressions –‐ Object Union Of”.

**Comment by rcrswld** [comment]
@konierik, @jdsika, @MircoNierenz, what do you think of Nicolás' (@BugGreen) proposal for integrating the OpenLABEL scenario-tagging-ontology in our existing structure?

- @konierik, can you please check our modelling approach / syntax?
- @jdsika , what do you think about adding another owl concept (unionOf) to our envited-x/PLC-AAD vocabular?
- @MircoNierenz, maybe just fyi right now - we will ping you as soon as our SHACL shapes were ready for review

In parallel and as long as you have no doubts, we will prepare a branch with a complete ontology and SHACL shape.
